config/jobs/periodic/docker-in-docker/periodic-dind-build.yaml

```diff
@@ -0,0 +1,213 @@
+            echo "There is no docker daemon." 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+            exit 1
+          else
+            # get the env file and the dockertest repo and the latest built of containerd if we don't want to build containerd
+            echo "*** COS Bucket ***" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+            CONT_NAME=docker_s3_env
+            docker run --env S3_SECRET_AUTH --env PATH_LOG_PROWJOB -d -v /workspace:/workspace --privileged --name $CONT_NAME debian:bullseye /bin/bash -c "${PATH_SCRIPTS}/get_env.sh"
+            status_code="$(docker container wait $CONT_NAME)"
+            echo $status_code 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+
+            if [[ ${status_code} -ne 0 ]]
+            then
+              echo "The docker to get the env.list and the dockertest has failed." 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+              docker logs $CONT_NAME 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+              exit 1
+            fi
+
+            ls /workspace 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+            
+            if [[ -f env.list ]]
+            then
+              # check there are 3 env variables in env.list
+              cat env.list 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+              if grep -Fq "DOCKER_VERS" env.list && grep -Fq "CONTAINERD_VERS" env.list && grep -Fq "PACKAGING_REF" env.list
+              then 
+                echo "DOCKER_VERS, CONTAINERD_VERS, PACKAGING_REF are in env.list" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+              else 
+                echo "DOCKER_VERS, CONTAINERD_VERS and/or PACKAGING_REF are not in env.list" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+                exit 1
+              fi
+              set -o allexport
+              source env.list
+            else
+              echo "There is no env.list" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+            fi
+            
+            # generate the env-distrib.list
+            echo "*** * env-distrib.list * ***" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+            mkdir docker-ce-packaging
+            pushd docker-ce-packaging
+            git init
+            git remote add origin  https://github.com/docker/docker-ce-packaging.git
+            git fetch --depth 1 origin ${PACKAGING_REF}
+            git checkout FETCH_HEAD
+
+            make REF=${DOCKER_VERS} checkout
+            popd
+
+            # get the packages list in the env_distrib.list
+            echo DEBS=\"`cd docker-ce-packaging/deb && ls -1d debian-* ubuntu-*`\" > env-distrib.list
+            echo RPMS=\"`cd docker-ce-packaging/rpm && ls -1d centos-* fedora-*`\" >> env-distrib.list
+
+            rm -rf docker-ce-packaging
+
+            # check the env.list (versions of docker-ce, containerd and list of packages)
+            if [[ -f env-distrib.list ]]
+            # if there is env.list and env-distrib.list
+            then
+              cat env-distrib.list 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+              # check if there are two variables in env-distrib.list
```

Comment:
Do we need to check DEBS and RPMS in `env-distrib.list`? Are they not meant to be there for sure?

Reply:
As well as for the env.list, I can change that